examples/lock-app/efr32/src/LockManager.cpp

```diff
@@ -407,6 +445,165 @@ bool LockManager::SetCredential(chip::EndpointId endpointId, uint16_t credential
     return true;
 }
 
+DlStatus LockManager::GetWeekdaySchedule(chip::EndpointId endpointId, uint8_t weekdayIndex, uint16_t userIndex,
+                                         EmberAfPluginDoorLockWeekDaySchedule & schedule)
+{
+    if ((weekdayIndex < MIN_SCHEDULE_INDEX) || (weekdayIndex > MAX_SCHEDULE_PER_USER))
+    {
+        ChipLogError(Zcl, "Cannot get schedule - schedule index is out of range [endpoint=%d,index=%d]", endpointId, weekdayIndex);
+        return DlStatus::kFailure;
+    }
+    if ((userIndex < MIN_USER_INDEX) || (userIndex > DOOR_LOCK_MAX_USERS))
+    {
+        ChipLogError(Zcl, "Cannot get schedule - user index is out of range [endpoint=%d,index=%d]", endpointId, userIndex);
+        return DlStatus::kFailure;
+    }
+
+    // doorlock server checks for valid indices
+    uint8_t adjustedWeekDayIndex = weekdayIndex - 1;
```

Comment:
you need to `static_assert(MIN_SCHEDULE_INDEX >= 1)` 

These seem to index within arrays. Could we have helper range checks like `IsValidIndexInArray`?

This code could look like:

```cpp
VerifyOrReturnValue(weekDayIndex > 0, DlStatus::kFailure); // indices are one-indexed
VerifyOrReturnValue(userIndex > 0, DlStatus::kFailure); // indices are one-indexed

weekdayIndex--;
userIndex--;

VerifyOrReturnValue(IsValidArrayIndex(userIndex, mWeekdaySchedules), DlStatus::kFailure);
VerifyOrReturnValue(IsValidArrayIndex(weekdayIndex, mWeekdaySchedules[userIndex]), DlStatus::kFailure);

schedule = mWeekdaySchedules[userIndex][weekdayIndex];
return DlStatus::kSuccess;
```

Reply:
Added helper functions, and tested them by disabling the range checking in door-lock-server temporarily.